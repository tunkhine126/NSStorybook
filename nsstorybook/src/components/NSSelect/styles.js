import { makeStyles } from '@material-ui/core/styles';
import colors from 'global-styles';

export const styles = (width = '100%', maxWidth) =>
  makeStyles({
    size: {
      width,
      maxWidth,
    },
    infoKey: {
      marginLeft: '5px',
    },
  });

export const useNSSelectStyle = (
  width,
  height,
  padding = '27px 12px 10px 15px'
) =>
  makeStyles(theme => ({
    root: {
      backgroundColor: colors.INTERFACE.ui1,
      transition: theme.transitions.create(['border-color']),
      width,
      height,
      padding,
    },
  }));
