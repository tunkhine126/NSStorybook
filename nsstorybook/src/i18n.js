/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
const addLocaleData = require('react-intl').addLocaleData; //eslint-disable-line
const enLocaleData = require('react-intl/locale-data/en');
const esLocaleData = require('react-intl/locale-data/es');
const ptLocaleData = require('react-intl/locale-data/pt');

// Custom Locale-Data for Haitian / Creole (not supported by react-intl)
const htLocaleData = {
  locale: 'ht',
  pluralRuleFunction: (e, t) => (t && e === 1 ? 'one' : 'other'),
};

const combineTranslationMessages = require('./translations/langs.json');

addLocaleData(enLocaleData);
addLocaleData(esLocaleData);
addLocaleData(ptLocaleData);
addLocaleData(htLocaleData);

const DEFAULT_LOCALE = 'en';

// prettier-ignore
const appLocales = [
  'en',
  'es',
  'pt',
  'ht'
];

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, combineTranslationMessages)
      : {};

  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key];
    return Object.assign(formattedMessages, {
      [key]: formattedMessage[locale],
    });
  };
  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

const translationMessages = {
  en: formatTranslationMessages('en', combineTranslationMessages),
  es: formatTranslationMessages('es', combineTranslationMessages),
  pt: formatTranslationMessages('pt', combineTranslationMessages),
  ht: formatTranslationMessages('ht', combineTranslationMessages),
};

exports.appLocales = appLocales;
exports.formatTranslationMessages = formatTranslationMessages;
exports.translationMessages = translationMessages;
exports.DEFAULT_LOCALE = DEFAULT_LOCALE;
