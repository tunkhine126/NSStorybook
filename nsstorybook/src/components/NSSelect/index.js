/**
 *
 * NSSelect
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import OutlinedInput from '@material-ui/core/OutlinedInput';

import colors from 'global-styles';
import { styles, useNSSelectStyle } from './styles';

function NSSelect({
  values,
  items,
  label,
  update,
  code = 'value',
  value = 'value',
  infoKey,
  width,
  maxWidth = '400px',
  outterWidth,
  outerHeight,
  padding,
  disableLabel,
  selectOverride,
  placeholderText,
  required = false,
  translate = '14px, 10px',
  className,
  onBlur = () => {},
}) {
  const classes = styles(width, maxWidth)();
  const labelFormatted = label.replace(/ /g, '-').toLowerCase();

  const menuItems = items.map(item => {
    if (item[infoKey]) {
      return (
        <MenuItem key={item[code]} value={item[code]}>
          {item[value]}
          <strong className={classes.infoKey}>({item[infoKey]})</strong>
        </MenuItem>
      );
    }
    return (
      <MenuItem key={item[code]} value={item[code]}>
        {item[value]}
      </MenuItem>
    );
  });

  const ComponentStyle = styled.div`
    .MuiInputLabel-outlined.MuiInputLabel-shrink {
      transform: translate(${translate}) scale(0.75);
    }
    .MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline {
      border-color: ${colors.STATE.active};
    }
    .MuiOutlinedInput-notchedOutline {
      border-color: ${colors.INTERFACE.ui4};
    }
    .MuiSvgIcon-root {
      color: ${colors.TEXT.light};
    }
    .MuiInputLabel-outlined {
      color: ${colors.TEXT.light};
      width: 77%;
    }
    .MuiOutlinedInput-root {
      padding: 1px;
    }
    .MuiInputLabel-outlined {
      ${disableLabel ? 'display: none;' : ''}
    }
  `;

  return (
    <ComponentStyle>
      <FormControl
        variant="outlined"
        className={classes.size}
        required={required}
      >
        {!placeholderText && (
          <InputLabel htmlFor={`${labelFormatted}-selector`}>
            {label}
          </InputLabel>
        )}
        <Select
          value={values.item || values}
          onChange={update}
          onBlur={onBlur}
          classes={selectOverride}
          className={className}
          displayEmpty={placeholderText}
          inputProps={{
            classes: useNSSelectStyle(outterWidth, outerHeight, padding)(),
          }}
          input={
            <OutlinedInput
              name={labelFormatted}
              id={`${labelFormatted}-selector`}
            />
          }
        >
          {placeholderText && <MenuItem value="">{label}</MenuItem>}
          {menuItems}
        </Select>
      </FormControl>
    </ComponentStyle>
  );
}

NSSelect.propTypes = {
  values: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
  items: PropTypes.array.isRequired,
  update: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
  onBlur: PropTypes.func,
  required: PropTypes.bool,
  code: PropTypes.string,
  value: PropTypes.string,
  placeholderText: PropTypes.bool,
  disableLabel: PropTypes.bool,
  infoKey: PropTypes.string,
  width: PropTypes.string,
  maxWidth: PropTypes.string,
  outterWidth: PropTypes.string,
  outerHeight: PropTypes.string,
  padding: PropTypes.string,
  translate: PropTypes.string,
  selectOverride: PropTypes.object,
};

export default NSSelect;
