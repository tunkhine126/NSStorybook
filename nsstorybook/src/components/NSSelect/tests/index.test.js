/**
 *
 * Tests for NSSelect
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import NSSelect from '../index';
import { DEFAULT_LOCALE } from '../../../../i18n';

describe('<NSSelect />', () => {
  const languages = [
    { code: 'ab', name: 'Abkhaz', nativeName: 'аҧсуа' },
    { code: 'aa', name: 'Afar', nativeName: 'Afaraf' },
    { code: 'af', name: 'Afrikaans', nativeName: 'Afrikaans' },
  ];

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <NSSelect
          values={{ item: 'us_en' }}
          items={languages}
          update={() => {}}
          label="Message"
          code="name"
          value="nativeName"
        />
      </IntlProvider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <NSSelect
          values={{ item: 'us_en' }}
          items={languages}
          update={() => {}}
          label="Message"
          code="name"
          value="nativeName"
        />
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
