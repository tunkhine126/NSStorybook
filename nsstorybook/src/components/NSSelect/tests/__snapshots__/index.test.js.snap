// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NSSelect /> Should render and match the snapshot 1`] = `
<div
  class="NSSelect__ComponentStyle-sc-1pe1cg7-0 ikAthF"
>
  <div
    class="MuiFormControl-root makeStyles-size-88"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink="true"
      for="message-selector"
    >
      Message
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby=" mui-component-select-message"
        class="MuiSelect-root makeStyles-root-90 MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="mui-component-select-message"
        role="button"
        tabindex="0"
      >
        <span>
          ​
        </span>
      </div>
      <input
        id="message-selector"
        name="message"
        type="hidden"
        value="us_en"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-171 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-172"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
